Update the new generator with the js-clients-shared imports

apinkert opened the pull request:
The way I've been testing these changes is by creating a new client (`npm run create-client`), changing the scripts in the new client's `package.json` to match the scripts in either `notifications-client` or `integrations-client` (since they're on the new generator) and running `npm run generate:prod` in the new client to generate all the API endpoints. All the `index.ts` files for the new endpoints should have the correct imports from the new `javascript-clients-shared` [client](https://github.com/RedHatInsights/javascript-clients/pull/234). One uncertainty I have about these new imports is the use of `BASE_PATH` in `utilBaseApi.mustache` no longer existing since we created a generic `base.api` in `javascript-clients-shared` to use, but I'm not sure `BASE_PATH` needs to be imported in all those endpoints since the new usage for clients will be for example:
```
const some-clientApi = APIFactory(BASE_PATH, undefined, { someEndpoint, someOtherEndpoint });
```
where BASE_PATH is already defined in a constants file.

Hyperkid123 commented:
`BASE_PATH` is an optional configuration. If non is set, the current browser origin will be used instead.

apinkert commented:
@Hyperkid123 I've added the overloads in. I've separated out `APIFactory` into its own `utils.ts` and moved pretty much all exported interfaces and types into `common.ts`
